src/bundle.ts

```diff
@@ -101,30 +82,30 @@ export class Bundler extends Transform {
   }
 
   isEntrypoint(file: File): boolean {
-    return this.allFragments.indexOf(file.path) !== -1;
+    return this.config.allFragments.indexOf(file.path) !== -1;
   }
 
   _buildBundles(): Promise<Map<string, string>> {
     return this._getBundles().then((bundles) => {
-      let sharedDepsBundle = (this.shell)
-          ? urlFromPath(this.root, this.shell)
+      let sharedDepsBundle = (this.config.shell)
+          ? urlFromPath(this.config.root, this.config.shell)
           : this.sharedBundleUrl;
       let sharedDeps = bundles.get(sharedDepsBundle) || [];
       let promises: Promise<any>[] = [];
 
-      if (this.shell) {
-        let shellFile = this.analyzer.getFile(this.shell);
+      if (this.config.shell) {
+        let shellFile = this.analyzer.getFile(this.config.shell);
         console.assert(shellFile != null);
         let newShellContent = this._addSharedImportsToShell(bundles);
         shellFile.contents = new Buffer(newShellContent);
       }
 
-      for (let fragment of this.allFragments) {
-        let fragmentUrl = urlFromPath(this.root, fragment);
-        let addedImports = (fragment === this.shell && this.shell)
+      for (let fragment of this.config.allFragments) {
+        let fragmentUrl = urlFromPath(this.config.root, fragment);
+        let addedImports = (fragment === this.config.shell && this.config.shell)
```

Comment:
for clarity let's move the `fragment === this.config.shell && this.config.shell` into a variable named `isShell`